var/spack/repos/builtin/packages/py-numba/package.py

```diff
@@ -12,10 +12,14 @@ class PyNumba(PythonPackage):
     homepage = "https://numba.pydata.org/"
     url      = "https://pypi.io/packages/source/n/numba/numba-0.35.0.tar.gz"
 
+    version('0.40.1', sha256='52d046c13bcf0de79dbfb936874b7228f141b9b8e3447cc35855e9ad3e12aa33')
     version('0.35.0', '4f447383406f54aaf18ffaba3a0e79e8')
 
+    depends_on('python@3.3:3.7.9999', type=('build', 'run'), when='@0.40.1:')
+    depends_on('python@3.3:3.6.9999', type=('build', 'run'), when='@:0.35.0')
     depends_on('py-numpy@1.10:',    type=('build', 'run'))
-    depends_on('py-llvmlite@0.20:', type=('build', 'run'))
+    depends_on('py-llvmlite@0.25:', type=('build', 'run'), when='@0.40.1:')
```

Comment:
I would not keep those open-ended for `0.X` software because it will likely break a lot.
One can add a comment that an "artificial" upper limit is just the latest tested working release of it.

Reply:
did you get burned by that? Personally, I'd rather prefer a breaking dependency over a hidden one...

Reply:
Yep, e.g. got many reports for building GCC by users. #9513 

Yep, somehow we need a way to document that we artificially added the latest *tested* release as upper range if we do this... but otherwise we just offload beta-testing to our users (when dependencies get updated and new defaults are due to that taken.)